sale_commission_agent_restrict/models/res_partner.py

```diff
@@ -35,3 +35,19 @@ def create(self, vals):
             vals["agent_ids"] = [(4, self.env.user.partner_id.id, 0)]
         res = super(ResPartner, self).create(vals)
         return res
+
+    def _update_fields_values(self, fields):
+        res = super(ResPartner, self)._update_fields_values(fields)
+        for agent in self.agent_ids:
+            user_id = self.env["res.users"].search([("partner_id", "=", agent.id)])
```

Comment:
That's better to use `user_id = agent.user_ids` as it benefits from auto_join and avoid to do a search per agent.

See https://github.com/odoo/odoo/blob/14.0/odoo/addons/base/models/res_partner.py#L219

Reply:
ok